### ISSUE TITLE
Accessibility support (a11y)

### ISSUE THREAD

**Issue by bbondy** [brave/browser-laptop#1346]
http://html5accessibility.com/

Cross post from the old repo:
https://github.com/brave/browser-for-gecko/issues/61

**Comment by luixxiul** [comment]
The link is 404 error.

**Comment by diracdeltas** [comment]
The old issue was just a link to https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA

**Comment by bbondy** [comment]
I asked a friend at Khan Academy (https://twitter.com/jdan) a while ago for his advise.  He is the author of http://khan.github.io/tota11y/ as well so maybe it will help us.  He has done React w/ a11y and he had this to say:

<img width="366" alt="screenshot 2016-08-04 21 41 16" src="https://cloud.githubusercontent.com/assets/831718/17423828/19b8d5c0-5a8c-11e6-8b68-a56987414458.png">

> You'd get a lot of quick wins in the Brave UI if your `<Button>` components rendered a `<button>` instead of a `<span>`

> Apr 13
>  Jordan Scales
>  This means you'll be able to tab around the brave UI with tab and shift+tab, then activate (or "click") items with enter or space. The next step would then be to provide these buttons with proper labels using something like `aria-label={this.props.label}` on the button component. This means that a screen-reader would read off "Refresh" after the user tabbed to the button. Unfortunately the "title" attribute won't cut it. (Likewise, "placeholder" on the URL bar is not good enough for a screen-reader)

> Apr 13
>  Jordan Scales
>  After that I started poking around but got 0 feedback from the `<webview>` :( My screenreader (VoiceOver on OSX) won't read a thing. This discourse posts (https://discuss.atom.io/t/electron-crash-with-webview-and-screen-reader-voiceover-jaws-and-nvda/23302 …) mentions that webview is exposing the accessibility tree on version 0.35.2 - is your fork of electron up to date with that? (this post is from December 2015)

> Apr 13
>  Jordan Scales
>  Beyond that there are a couple more subtle things to keep in mind. The tabs should follow "ARIA best practices" so that a blind user would know they're dealing with actual tabs. There are patterns outlined in super complex documents like https://www.w3.org/TR/wai-aria-practices/#tabpanel … (or, more practically something like http://heydonworks.com/practical_aria_examples/#tab-interface …). You'll also need to make you alert the user when a new tab or window opens. I've never done accessibility at the browser level so I don't have any experience with this - but it's an interesting case since the whole thing is written with javascript!
> Apr 13
>  Jordan Scales
>  I'm not familiar with electron but if `<webview>` itself is not accessible to screen-readers then I'm afraid you'll be stuck.

Of course we can always reach into webviews in our front end code, or even drop down to electron and chromium level code if needed, so I suspect we aren't that stuck even if the situation hasn't improved already with webviews.  The webviews are only used for the web content though and not our browser UI.

**Comment by luixxiul** [comment]
Also from support:
> it did not seem to be accessible with Jaws for Windows, a screen reader for the blind. As you may know, JAWS is one of the main pieces of assistive technology the blind use on a PC.

**Comment by jkup** [comment]
Do you think it would be easier to start breaking these out into separate tickets? Switching button.js to output <buttons> seems like an easy win!

**Comment by bbondy** [comment]
yes this whole issue is better as a label, once you post things that are contained within her feel free to close, thanks.  There will be ongoing work here.

**Comment by westes** [comment]
Under Ubuntu gnome + orca, no controls are read, although the title of the app window is read out.

**Comment by bsclifton** [comment]
Closing as `wontfix` since HTML5 accessibility (for the UI) will no longer be an issue when moving to [Brave Core](https://github.com/brave/brave-browser) (which uses C++ for the UI)
